File: service/src/main/java/hu/bsstudio/scores/qualification/configuration/QualificationScoreServiceConfiguration.java
package hu.bsstudio.scores.qualification.configuration;

import hu.bsstudio.repository.TeamRepository;
import hu.bsstudio.scores.qualification.DefaultQualificationScoreService;
import hu.bsstudio.scores.qualification.QualificationScoreService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class QualificationScoreServiceConfiguration {

    @Autowired
    private TeamRepository repository;

    @Bean
    public QualificationScoreService qualificationScoreService() {
        return new DefaultQualificationScoreService(repository);
    }
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/scores/qualification/configuration/QualificationScoreServiceConfiguration.kt
package hu.bsstudio.robonaut.scores.qualification.configuration

import hu.bsstudio.robonaut.repository.TeamRepository
import hu.bsstudio.robonaut.scores.qualification.BroadcastingQualificationScoreService
import hu.bsstudio.robonaut.scores.qualification.DefaultQualificationScoreService
import hu.bsstudio.robonaut.scores.qualification.QualificationScoreService
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class QualificationScoreServiceConfiguration(
    private val rabbitTemplate: RabbitTemplate,
    private val repository: TeamRepository,
) {
    @Bean
    fun qualificationScoreService(defaultQualificationScoreService: QualificationScoreService): QualificationScoreService =
        BroadcastingQualificationScoreService(rabbitTemplate, defaultQualificationScoreService)

    @Bean
    fun defaultQualificationScoreService(): QualificationScoreService = DefaultQualificationScoreService(repository)
}
